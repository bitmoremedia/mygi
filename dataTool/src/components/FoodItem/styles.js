import styled from 'styled-components';

export const Item = styled.div`
  padding: 5px;
  white-space: nowrap;
`;
