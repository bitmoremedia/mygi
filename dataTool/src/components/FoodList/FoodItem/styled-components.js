import styled from 'styled-components'

export const Item = styled.div`
  padding: 5px;
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  justify-content: space-between
`

export const FoodName = styled.div`
  white-space: nowrap;
`

export const EditButton = styled.div`
  margin-left: 10px;
`
