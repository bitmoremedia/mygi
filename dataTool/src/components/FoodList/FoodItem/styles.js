import styled from 'styled-components';

export const Item = styled.div`
  padding: 5px;
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  justify-content: space-between
`;

export const FoodName = styled.div`
  white-space: nowrap;
`;

export const EditButton = styled.button`
  margin-left: 10px;
`;
