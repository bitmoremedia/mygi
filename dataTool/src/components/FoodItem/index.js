import React from 'react';

import { Item } from './styles';

const FoodItem = ({food}) => {
  return (
    <Item key={`${food.id}-food-item`}>
      {food.name}
    </Item>
  );
};

export default FoodItem;
