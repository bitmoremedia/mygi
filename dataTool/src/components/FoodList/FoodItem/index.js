import React, { PropTypes } from 'react';

import { Item } from './styles';

const FoodItem = ({food, editAction}) => {
  return (
    <Item key={`${food.id}-food-item`}>
      {food.name}
    </Item>
  );
};

FoodItem.propTypes = {
  food: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    gi: PropTypes.any.isRequired,
  }).isRequired,
};

export default FoodItem;
