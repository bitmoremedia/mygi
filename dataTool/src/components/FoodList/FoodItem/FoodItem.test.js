import React from 'react'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'

import FoodItem from './index'

const food = {
  id: 'abc-123',
  name: 'Kidney Beans',
  gi: 36,
}

test('component renders without error', () => {
  const wrapper = mount(
    <FoodItem
      food={food}
      editAction={() => {}}
    />,
  )
  expect(wrapper.exists()).toBe(true)
})

// SNAPSHOT TESTS

test('component renders as expected', () => {
  const wrapper = mount(
    <FoodItem
      food={food}
      editAction={() => {}}
    />,
  )
  expect(toJson(wrapper)).toMatchSnapshot()
})
