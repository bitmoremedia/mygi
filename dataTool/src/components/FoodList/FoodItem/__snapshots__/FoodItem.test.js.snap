exports[`test component renders as expected 1`] = `
<Component
  editAction={[Function]}
  food={
    Object {
      "gi": 36,
      "id": "abc-123",
      "name": "Kidney Beans",
    }
  }>
  <styled.div>
    <div
      className="gpERpa">
      <styled.div>
        <div
          className="egTnmM">
          Kidney Beans
        </div>
      </styled.div>
      <styled.div>
        <div
          className="jTDtNq">
          <Component
            danger={false}
            onClick={[Function]}>
            <styled.button
              onClick={[Function]}>
              <button
                className="OgMJe"
                onClick={[Function]}>
                edit
              </button>
            </styled.button>
          </Component>
        </div>
      </styled.div>
    </div>
  </styled.div>
</Component>
`;
